### ISSUE TITLE
Server-side hashing, url sanitation, redirecting logged in users to home 

### ISSUE THREAD

**Issue by k-hendricks** [taa176/web#6]
This is what I'm planning to work on today. I will also be working on https implementation and server symmetric encryption, but I probably won't get those finished.

**Comment by k-hendricks** [comment]
Oh I am also going to do try and handle Neitzsche-sanitation. He seems to have appeared, like the first pustule of plague, on our homepage.

**Comment by taa176** [comment]
I'm still wondering about the server encryption. Almost all references I find suggest using different salts for each user and hashing server-side. Is encryption a good enough substitute? Is there a good reference that talks about this?

There are still some things I have questions about, including how we will secure the private key. Hashing each password also has the benefit that you have to `crack` each password individually if you want to get any information. W/ encryption, if the attacker has the key, they'll have access to the entire database.

To your other point (note the spelling): 
This is a Nietzsche website. You're welcome.

**Comment by k-hendricks** [comment]
Yeah storing the key in somewhere more secure than the server is difficult.
It would just be nice because it would protect more than just the
passwords.
I'm not proposing that server side encryption be used to protect plaintext
passwords. Remember they would still be hashed clientside with the username
+ constant random string as salt. This is only to prevent the server from
accepting requests if someone gets access to the hashed password and the
server is directly comparing that to the hash on file. I can't really think
of a situation where an attacker could dump the entire database and decrypt
it, but not be able to just modify the server code to accept "opensesame"
as a password directly in plaintext.
Still its better to be safe so I will add server side hashing in the normal
way with random salts.

**Comment by taa176** [comment]
Okay, sounds good to me. Facebook should hire us as consultants

https://newsroom.fb.com/news/2019/03/keeping-passwords-secure/

**Comment by k-hendricks** [comment]
Lol poor people need not worry. These guys are terrible.
"Facebook Lite is a version of Facebook predominantly used by people in
regions with lower connectivity."
